src/api/RestAPI.ts

```diff
@@ -302,44 +322,22 @@ export class RestAPI extends EventEmitter {
      * @internal
      */
     async longPolling<Response>(
-        /**
-         * API call to be repeated
-         */
         promise: PromiseCreator<Response>,
-
-        /**
-         * Condition for which the response is considered to be successful and stop the polling
-         */
-        successCondition: (response: Response) => boolean,
-
-        /**
-         * Condition to cancel the polling without triggering error
-         */
-        cancelCondition?: (response: Response) => boolean,
-
-        /**
-         * Callback to be triggered at the end of the long polling
-         */
-        callback?: ResponseCallback<Response>,
-
-        /**
-         * Time after which a TimeoutError will be triggered and the poll will be canceled
-         */
-        timeout: number = POLLING_TIMEOUT,
-
-        /**
-         * Interval between polls
-         */
-        interval: number = POLLING_INTERVAL,
-
-        /**
-         * Callback to be triggered at each poll iteration
-         */
-        iterationCallback?: (response: Response) => void
+        pollParams: PollParams<Response>
     ): Promise<Response> {
+        const {
+            successCondition,
+            cancelCondition,
+            iterationCallback,
+            timeout = POLLING_TIMEOUT,
+            interval = POLLING_INTERVAL,
+        } = pollParams;
+
         return execRequest(async () => {
             let internalErrorCount = 0;
 
+            const sleepInterval = () => new Promise((resolve) => setTimeout(resolve, interval));
+
             const repeater = async (): Promise<Response> => {
                 try {
                     const result = await promise();
```

Comment:
maybe `promise` could be renamed as `task`? `PromiseCreator` -> `Task`?

Reply:
Seems that other part of the application call it `executor`. Let me rename it this way for consistency. Thanks for catching this.